### PR TITLE
Patch holes in reference generation

### PR DESCRIPTION
**Pull request by thejcannon**
Fixes most of https://github.com/pantsbuild/pantsbuild.org/issues/37 (sans 1, really) with some additional changes to the templates and generator.

### CODE REVIEW

**Comment by thejcannon** [comment]
This PR contains commits for:

- Re-running the reference generation against 2.19 (which previously used the Python/Jinja template): [`0433c5e` (#53)](https://github.com/pantsbuild/pantsbuild.org/pull/53/commits/0433c5ef5c67c28dbad41300b6d715bee3c1be8a)
- Re-running the reference generation against `docs` (which used the previous version of the JS/Mustache template): [`85d1ac4` (#53)](https://github.com/pantsbuild/pantsbuild.org/pull/53/commits/85d1ac44f12d9ab5bf49750347812bd98723ff38) 
- Undo-ing those commits

That way you can compare this change against the baseline (`main`) and the prior art (`2.19`).

After this change goes in, we can re-generate the reference docs for every version.

**Comment by thejcannon** [comment]
We should still fix: https://github.com/pantsbuild/pantsbuild.org/pull/50#discussion_r1440102814

File: reference_codegen/generate.js [hunk 1]
```diff
@@ -178,13 +196,13 @@ Object.entries(helpAll.name_to_target_type_info).forEach(([name, info]) => {
...
 fs.writeFileSync(
   "subsystems/_category_.json",
-  '{\n  "label": "Subsystems",\n  "link": {\n    "type": "generated-index",\n    "slug": "/reference/subsystems",    "title": "Subsystems"\n  }}\n'
+  '{\n  "label": "Subsystems",\n  "link": {\n    "type": "generated-index",\n    "slug": "/reference/subsystems",\n    "title": "Subsystems"\n  }\n}\n'
 );
 fs.writeFileSync(
   "targets/_category_.json",
-  '{\n  "label": "Targets",\n  "link": {\n    "type": "generated-index",\n    "slug": "/reference/targets",\n    "title": "Targets"\n  }}\n'
+  '{\n  "label": "Targets",\n  "link": {\n    "type": "generated-index",\n    "slug": "/reference/targets",\n    "title": "Targets"\n  }\n}\n'
 );
```
Comment (huonw) [inline reference_codegen/generate.js:208]: Just a minor style thing (no change required): We could potentially use template literals (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#multi-line_strings) to have these written out more literally and thus more obvious contents. Either that, or separate JSON file templates that can be copied verbatim.

**Review by huonw — APPROVED** [review]
Nice

File: reference_codegen/generate.js (hunk 1 above)
Comment (thejcannon) [reply reference_codegen/generate.js:208]: Ah I didn't know JS template literals preserved newlines.

Yeah feel free to PR id +1 that

**Pull request merged**